### ISSUE TITLE
CA2007 False Negative when a local function is async and an outer method isn't async

### ISSUE THREAD

**Issue by Tan90909090** [dotnet/roslyn-analyzers#2393]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.2

#### Diagnostic ID

[CA2007](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2007-do-not-directly-await-task)

#### Repro steps
```cs
using System.Threading.Tasks;
class C
{
    Task SomeAsync()
    {
        return CoreAsync();

        async Task CoreAsync()
        {
            await Task.Delay(1); // CA2007 should be reported
        }
    }
}
```

#### Expected behavior
CA2007 `Do not directly await a Task without calling ConfigureAwait` is reported on `await Task.Delay(1);`

#### Actual behavior
CA2007 is not reported.

Note that if the outer method is async then CA2007 is reported as expected:
```cs
using System.Threading.Tasks;
class C
{
    async Task SomeAsync()
    {
        await CoreAsync(); // CA2007 is reported here

        async Task CoreAsync()
        {
            await Task.Delay(1); // also CA2007 is reported here
        }
    }
}
```
